iktomi/cli/app.py

```diff
@@ -69,17 +61,35 @@ def command_serve(self, host='', port='8000', level='debug'):
             server_thread.start()
 
             wait_for_code_change(extra_files=self.extra_files)
-
             server_thread.running = False
             server_thread.join()
             logger.info('Reloading...')
-
-            # Smart reload of current process.
-            # Main goal is to reload all modules
-            # NOTE: For exec syscall we need to flush and close all fds manually
             flush_fds()
-            close_fds()
-            os.execvp(sys.executable, [sys.executable] + sys.argv)
+            pid = os.fork()
+            # We need to fork before `execvp` to perform code reload
+            # correctly, because we need to complete python destructors and
+            # `atexit`.
+            # This will save us from problems of incorrect exit, such as:
+            # - unsaved data in data storage, which does not write data
+            # on hard drive immediatly
+            # - uncovered code with coverage utility, which uses `atexit`
+            # handlers in its working
+            # NOTE: we using untipical fork-exec scheme with replacing
+            # the parent process, not the child
+            if pid: # pragma: no cover
+                # parent process cannot be covered with coverage, because it is
```

Comment:
It's not clear that this comment corresponds to `pragma` above. Is it possible to make it obvious?